src/Microsoft.PowerShell.ConsoleGuiTools/OutConsoleGridviewCmdletCommand.cs

```diff
@@ -66,6 +66,16 @@ public SwitchParameter PassThru
         // This method gets called once for each cmdlet in the pipeline when the pipeline starts executing
         protected override void BeginProcessing()
         {
+            if (Console.IsInputRedirected)
+            {
+                ErrorRecord error = new ErrorRecord(
+                    new PSNotSupportedException("Not supported in this environment (when input is redirected)."),
```

Comment:
Not part of this PR, but it appears msbuild now supports resgen, so we should move strings to a resx file.  See https://github.com/microsoft/msbuild/issues/2272#issuecomment-532268770

Reply:
we should do this across all of our repos. Would be nice if someone did the ground work on how to support this and then taught everyone else.

Reply:
We have the tracking issue https://github.com/PowerShell/PowerShell/issues/2882